File: back-end-gc-main/src/main/java/com/gi/gestioncompetence/entity/Competence.java
package com.gi.gestioncompetence.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
public class Competence {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long idCompetence;

  private String nomCompetence;
  private String descriptionCompetence;
  private String categorieDomaine;
  private String niveauCompetence;

}

File: front-end-gc-main/src/app/entities/competence.ts
export class Competence {
    idCompetence: number;
    nomCompetence: string;
    descriptionCompetence: string;
    categorieDomaine: string;
    niveauCompetence: string;
  
    constructor(
      idCompetence: number,
      nomCompetence: string,
      descriptionCompetence: string,
      categorieDomaine: string,
      niveauCompetence: string
    ) {
      this.idCompetence = idCompetence;
      this.nomCompetence = nomCompetence;
      this.descriptionCompetence = descriptionCompetence;
      this.categorieDomaine = categorieDomaine;
      this.niveauCompetence = niveauCompetence;
    }
  }
  